File: planik-android/app/src/main/java/pl/planik/domain/model/Day.kt
package pl.planik.domain.model

import java.time.DayOfWeek
import java.time.OffsetDateTime

data class Day(
  val ordinal: Int,
  val dayOfWeek: DayOfWeek,
  val date: OffsetDateTime,
  val entries: List<DayEntry>
)

File: planik-flutter/planik/lib/domain/model/day.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:planik/domain/model/day_entry.dart';

part 'day.freezed.dart';

@freezed
class Day with _$Day {
  const factory Day({
    required int ordinal,
    required String name,
    required DateTime date,
    required List<DayEntry> entries,
  }) = _Day;
}
